shared/modules/sizeof.js

```diff
@@ -18,91 +18,73 @@ const ECMA_SIZES = {
+    case 'boolean':
+      return ECMA_SIZES.BOOLEAN;
+    case 'number':
+      return ECMA_SIZES.NUMBER;
+    case 'symbol': {
+      const isGlobalSymbol = Symbol.keyFor && Symbol.keyFor(data);
+      return isGlobalSymbol
+        ? Symbol.keyFor(data).length * ECMA_SIZES.STRING
+        : (data.toString().length - 8) * ECMA_SIZES.STRING;
     }
+    default:
+      throw new Error(`Unrecognized type '${dataType}'`);
   }
-  if (Object.getOwnPropertySymbols) {
-    const symbolProperties = Object.getOwnPropertySymbols(obj);
-    Array.prototype.push.apply(stringProperties, symbolProperties);
-  }
-  return stringProperties;
 }
 
-function sizeOfObject(state, object) {
-  if (object === null) {
+/**
+ * Main module's entry point
+ * Calculates Bytes for the provided parameter
+ *
+ * @param data - handles object/string/boolean/buffer
+ * @param maxSize
+ * @returns {*}
+ */
+export function assertObjectMaxSize(data, maxSize = 0) {
+  let size = 0;
+  const topLevelType = typeof data;
+  if (topLevelType !== 'object') {
+    size = getPrimitiveSize(data, topLevelType);
+    if (size > maxSize) {
+      throw new Error('object exceeded max size');
+    }
     return;
   }
 
-  const properties = allProperties(object);
-  const calc = getCalculator(state);
-  for (let i = 0; i < properties.length; i++) {
-    const key = properties[i];
-    calc(key);
-    calc(object[key]);
-  }
-}
+  const objects = [data];
 
-function getCalculator(state) {
-  return function calculator(object) {
-    if (state.size > state.maxSize) {
-      throw new Error('object exceeded max size');
-    }
-    if (Buffer.isBuffer(object)) {
-      state.size += object.length;
+  function countValueSize(value) {
+    const valueType = typeof value;
+    if (valueType === 'object') {
+      objects.push(value);
+    } else {
+      size += getPrimitiveSize(value, valueType);
+      if (size > maxSize) {
+        throw new Error('object exceeded max size');
+      }
     }
+  }
 
-    const objectType = typeof object;
-    switch (objectType) {
-      case 'string':
-        // https://stackoverflow.com/questions/68789144/how-much-memory-do-v8-take-to-store-a-string/68791382#68791382
-        state.size += isNodePlatform
-          ? 12 + 4 * Math.ceil(object.length / 4)
-          : object.length * ECMA_SIZES.STRING;
-        return;
-      case 'boolean':
-        state.size += ECMA_SIZES.BOOLEAN;
-        return;
-      case 'number':
-        state.size += ECMA_SIZES.NUMBER;
-        return;
-      case 'symbol': {
-        const isGlobalSymbol = Symbol.keyFor && Symbol.keyFor(object);
-        state.size += isGlobalSymbol
-          ? Symbol.keyFor(object).length * ECMA_SIZES.STRING
-          : (object.toString().length - 8) * ECMA_SIZES.STRING;
-        return;
+  for (const object of objects) {
+    if (Array.isArray(object)) {
+      for (const value of object) {
+        countValueSize(value);
+      }
+    } else {
+      for (const key of Object.keys(object)) {
```

Comment:
```suggestion
      // Iterate through keys first because they're faster to count, minimizing the number of object references we need to track before stopping
      for (const key of Object.keys(object)) {
```